src/editor/Editor.jsx

```diff
@@ -1,293 +1,344 @@
 import React from 'react';
+import ReactDOM from 'react-dom';
 import Classable from "../mixins/classable";
-import Toolbar from "../Toolbar/Toolbar";
-import ToolbarGroup from "../Toolbar/ToolbarGroup";
-import FontIcon from "../FontIcon/FontIcon";
-import IconButton from "../Button/IconButton";
-import DropDownIcon from "../DropDownIcon";
-import Dialog from "../Dialog/Dialog";
-import TextField from "../Input/TextField";
-import ColorPicker from "../ColorPicker/ColorPicker";
-import ContentRte from "./ContentRte";
-import ContentSrc from "./ContentSrc";
-
-var fontStyleOptions = [
-	{ payload: '<p>', text: 'Body' },
-	{ payload: '<h1>', text: 'Heading 1' },
-	{ payload: '<h2>', text: 'Heading 2' },
-	{ payload: '<h3>', text: 'Heading 3' },
-	{ payload: '<h4>', text: 'Heading 4' }
+import immutable from 'immutable';
+import classnames from 'classnames';
+import ThemeService from '../styles/ChamelThemeService';
+import {Editor, EditorState, RichUtils} from 'draft-js';
+
+// Custom overrides for "code" style.
+const styleMap = {
+    CODE: {
+        backgroundColor: 'rgba(0, 0, 0, 0.05)',
+        fontFamily: '"Inconsolata", "Menlo", "Consolas", monospace',
+        fontSize: 16,
+        padding: 2,
+    },
+};
+
+// Block Types that are used in the editor and displayed in the toolbar
+const BLOCK_TYPES = [
+    {label: 'H1', style: 'header-one'},
+    {label: 'H2', style: 'header-two'},
+    {label: 'H3', style: 'header-three'},
+    {label: 'H4', style: 'header-four'},
+    {label: 'H5', style: 'header-five'},
+    {label: 'H6', style: 'header-six'},
+    {label: 'Blockquote', style: 'blockquote'},
+    {label: 'UL', style: 'unordered-list-item'},
+    {label: 'OL', style: 'ordered-list-item'},
+    {label: 'Code Block', style: 'code-block'},
 ];
 
-var fontNameOptions = [
-	{ payload: 'Arial', text: 'Arial' },
-	{ payload: 'Georgia', text: 'Georgia' },
-	{ payload: 'Tahoma', text: 'Tahoma' },
-	{ payload: 'Courier New', text: 'Courier New' },
-	{ payload: 'Times New Roman', text: 'Times New Roman' },
-	{ payload: 'Verdana', text: 'Verdana' }
+// Inline Styles that are used in the editor and displayed in the toolbar
+const INLINE_STYLES = [
+    {label: 'Bold', style: 'BOLD'},
+    {label: 'Italic', style: 'ITALIC'},
+    {label: 'Underline', style: 'UNDERLINE'},
+    {label: 'Monospace', style: 'CODE'},
 ];
 
-var fontSizeOptions = [
-	{payload: 1, text: 'Smallest'},
-	{payload: 2, text: 'X-Small'},
-	{payload: 3, text: 'Small'},
-	{payload: 4, text: 'Normal'},
-	{payload: 5, text: 'Large'},
-	{payload: 6, text: 'X-Large'},
-	{payload: 7, text: 'Huge'}
-];
+/**
+ * Handle when displaying the block type controls and
+ * Set the styles of the controls if it is toggled or not
+ */
+const BlockStyleControls = (props) => {
```

Comment:
Great job using a pure function component. Those help make it faster. However, we should probably make this a separate file and import it to keep this class short.